File: 0946_validateStackSequences/main.py
from typing import List

class Solution:
    def validateStackSequences(self, pushed: List[int], popped: List[int]) -> bool:
        stack = []
        i = 0
        for val in pushed:
            stack.append(val)
            while i < len(popped) and stack and popped[i] == stack[-1]:
                stack.pop()
                i += 1
        return not stack

File: 0946_validateStackSequences/main.cpp
#include <vector>
#include <stack>
class Solution {
public:
    bool validateStackSequences(std::vector<int>& pushed, std::vector<int>& popped) {
        std::stack<int> st;
        int i = 0;
        for(int val: pushed) {
            st.push(val);
            while(i < popped.size() && !st.empty() && st.top() == popped[i]) {
                st.pop();
                ++i;
            }
        }
        return st.empty();
    }
};